docs/src/main/paradox/server/mtls.md

```diff
@@ -0,0 +1,7 @@
+# Mutual authentication with TLS
+
+Mutual or mTLS means that just like how a client will only connect to servers with valid certificates, the server will
+also verify the client certificate and only allow connections if the client key pair is accepted by the server. This is
+useful for example in microservices where only other services are allowed to interact with a service, and public access 
+should be denied.
+
```

Comment:
is the plan to include the `MtlsGreeterServer` as example here?
do we also need example of the client?

Reply:
Yes, that's the plan, with the client, I need to figure out how to do that first though :)